frontend/src/_helpers/utils.js

```diff
@@ -1060,3 +1060,48 @@ export const determineJustifyContentValue = (value) => {
       return 'start';
   }
 };
+
+export const defaultWhiteLabellingSettings = {
+  WHITE_LABEL_LOGO: 'https://app.tooljet.com/logo.svg',
+  WHITE_LABEL_TEXT: 'ToolJet',
+  WHITE_LABEL_FAVICON: 'https://app.tooljet.com/favico.png',
+};
+
+export const setWindowTitle = async (pageDetails, location) => {
+  const pathToTitle = {
+    'instance-settings': 'Settings',
+    'workspace-settings': 'Workspace settings',
+    integrations: 'Marketplace',
+    workflows: 'Workflows',
+    database: 'Database',
+    'data-sources': 'Data sources',
+    'audit-logs': 'Audit logs',
+    'account-settings': 'Profile settings',
+    settings: 'Profile settings',
+  };
+  const whiteLabelText = defaultWhiteLabellingSettings.WHITE_LABEL_TEXT;
+  let pageTitleKey = pageDetails?.page || '';
+  let pageTitle = '';
+  if (!pageTitleKey) {
+    pageTitleKey = Object.keys(pathToTitle).find((path) => location?.pathname?.includes(path)) || '';
```

Comment:
I think it is better to be more specific when we check a path against an URL. i think this flashing issue might be the cause of this change

https://github.com/ToolJet/ToolJet/assets/18279004/3ba075e6-7344-462e-ba7a-ca0a750069dc

Adding more details about the above video.
If the app slug contains any of these paths https://arc.net/l/quote/tfxzywll it will flash the path first then change to the app-name

Reply:
Resolved the issue, For handling apps and applications, I have added a separate check which will resolve the falshing issue. We have limited main pages on the main screen. I do not think they'll be sharing any of the keywords added in the object in the future as well. So, this should be full proof.

Reply:
FYI, Usually when we do paths check we specifically take that subpath and run against the paths array. Thats is more secure way to do.